plugins/inx/server_utxo.go

```diff
@@ -216,10 +233,92 @@ func (s *INXServer) ListenToLedgerUpdates(req *inx.LedgerUpdateRequest, srv inx.
 	return ctx.Err()
 }
 
+func (s *INXServer) ListenToTreasuryUpdates(req *inx.LedgerRequest, srv inx.INX_ListenToTreasuryUpdatesServer) error {
+	var sentTreasuryUpdate bool
+	sendPreviousMilestoneDiffs := func(startIndex milestone.Index) error {
+		deps.UTXOManager.ReadLockLedger()
+		defer deps.UTXOManager.ReadUnlockLedger()
+
+		ledgerIndex, err := deps.UTXOManager.ReadLedgerIndex()
```

Comment:
```suggestion
		ledgerIndex, err := deps.UTXOManager.ReadLedgerIndexWithoutLocking()
```

Reply:
Also look at the rest of the server code, there is at least another part where we did that mistake. (I can't mark the position because that line is not edited :trollface: )